### PR TITLE
Feature/Logic

### PR DESCRIPTION
**Pull request by EroAuditore**
- Import the files calculator.js and operate.js into a logic/
- Implement the event handlers you need to use the math logic from calculator.js and operate.js
- State and props, pass props to child components, lift state up.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,57 @@
...
-function App() {
-  return (
-    <div className="App">
-      <h1>My first react app</h1>
-    </div>
-  );
+import React, { Component } from 'react';
+import { Container } from 'react-bootstrap';
+import Calculator from './components/Calculator';
+import calculate from './Logic/calculate';
```
Comment (arslanbisharat) [inline src/App.js]: - [x] calculate.js should not be imported here. Please remove it from here and related code and let's move it to component/calcultor.js. Please have another look at the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/project_math_magicians_events/math-magicians/project_events.md).

File: src/components/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Button } from 'react-bootstrap';
+
+class Calculator extends Component {
```
Comment (arslanbisharat) [inline src/components/Calculator.js]: - [x] Please import `src/Logic/calculate.js `and `src/Logic/operate.js `here in `src/components/Calculator.js`.

File: src/components/Calculator.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+              className="w-100"
+              onClick={() => handleClick('=')}
+            >
+              =
+            </Button>
+          </Col>
+        </Row>
+      </>
+    );
+  }
```
Comment (arslanbisharat) [inline src/components/Calculator.js]: - [x] I cannot run a few operations such as percent. Please work on it.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [x] You have an empty file **src/components/tes.js**. Please remove it. 
- [x] Please follow the same design given in the requirements. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 3 above)
Comment (EroAuditore) [reply src/components/Calculator.js]: If you look at the code provided by microverse, the operation is not a percentage, is the mod operation 

https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js

Comment (arslanbisharat) [reply src/components/Calculator.js]: Yes but it does not work too Sir. I doesn't give any result.

Comment (EroAuditore) [reply src/components/Calculator.js]: For me is working
mod

Can you share the steps you are doing so I can reproduce your error and fix it?

Comment (arslanbisharat) [reply src/components/Calculator.js]: Please request a review and let the code reviewer see your project

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @EroAuditore  🔥 

You worked hard in all the instructions that the previous Code Reviewer gives you, your project is up and running👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by krys2fa — COMMENTED** [review]
Hello @EroAuditore 👋,

_This is an additional review as a result of the second opinion request._

Well done on making the changes requested and eventually getting the PR approved. Kudos for sticking with it and submitting it for another review. 
Congratulations once again on the approval.

Keep rocking 🚀!

**Pull request merged**